app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java

```diff
@@ -395,6 +395,31 @@ public boolean isInPrivateMode() {
         return mPrivateMode;
     }
 
+    public void enterImmersiveMode() {
+        for (WindowWidget window: mRegularWindows) {
+            if (window != mFocusedWindow) {
+                window.onPause();
+            }
+        }
+        for (WindowWidget window: mPrivateWindows) {
+            if (window != mFocusedWindow) {
+                window.onPause();
+            }
+        }
+    }
+
+    public void exitImmersiveMode() {
+        for (WindowWidget window: mRegularWindows) {
+            if (window != mFocusedWindow) {
+                window.onResume();
+            }
+        }
+        for (WindowWidget window: mPrivateWindows) {
+            if (window != mFocusedWindow) {
+                window.onResume();
+            }
+        }
```

Comment:
Aren't we pausing private window when in regular browsing mode. Won't this resume them when coming out of immersive mode?

Reply:
Good catch, fixed. We should only pause/resume de windows in the current mode.